cpp/cmake/Modules/SetGPUArchs.cmake

```diff
@@ -25,35 +25,41 @@ else()
   list(REMOVE_ITEM SUPPORTED_CUDA_ARCHITECTURES "62" "72")
 endif()
 
-if(CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11)
+# CMake < 3.20 has a bug in FindCUDAToolkit where it won't properly detect the CUDAToolkit version
+# when find_package(CUDAToolkit) occurs before enable_language(CUDA)
+if(NOT DEFINED CUDAToolkit_VERSION AND CMAKE_CUDA_COMPILER)
+  execute_process(COMMAND ${CMAKE_CUDA_COMPILER} "--version" OUTPUT_VARIABLE NVCC_OUT)
+  if(NVCC_OUT MATCHES [=[ V([0-9]+)\.([0-9]+)\.([0-9]+)]=])
+    set(CUDAToolkit_VERSION_MAJOR "${CMAKE_MATCH_1}")
+    set(CUDAToolkit_VERSION_MINOR "${CMAKE_MATCH_2}")
+    set(CUDAToolkit_VERSION_PATCH "${CMAKE_MATCH_3}")
+    set(CUDAToolkit_VERSION "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}")
+  endif()
+  unset(NVCC_OUT)
+endif()
+
+if(CUDAToolkit_VERSION_MAJOR LESS 11)
```

Comment:
Was there a reason for changing the `version_less` to `less` here?

Reply:
I assume because we're now comparing the `major` component of the version (i.e. 11) instead of the entire version number (i.e. 11.0.0), so we have to use `LESS` instead of `VERSION_LESS`.

I copied this over from [RMM's implementation](https://github.com/rapidsai/rmm/blob/branch-0.19/cmake/Modules/SetGPUArchs.cmake#L28-L49). I'm pretty sure `CMAKE_CUDA_COMPILER_VERSION` gets populated by `enable_language(CUDA)`, but now that happens after this module is included, so we have to switch all these to use the variables populated by `find_package(CUDAToolkit)`.